Chapter02/Exercise2.03/app/src/main/java/com/example/saveandrestore/MainActivity.kt

```diff
@@ -4,23 +4,49 @@ import android.content.Context
 import android.os.Bundle
 import android.util.Log
 import android.view.inputmethod.InputMethodManager
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
 import android.widget.Toast
 import androidx.appcompat.app.AppCompatActivity
-import kotlinx.android.synthetic.main.activity_main.*
 import java.util.*
 
 class MainActivity : AppCompatActivity() {
 
+    private val discountButton by lazy {
```

Comment:
@drulabs - This is the first time in the course that 'by lazy' has been seen. Could you update the text following the example in the chapter explaining this?

Reply:
changed to get() and added some explanation in the chapter